Source/Core/VideoBackends/OGL/TextureCache.cpp

```diff
@@ -119,12 +141,22 @@ TextureCache::TCacheEntryBase* TextureCache::CreateTexture(const TCacheEntryConf
 
   glTexParameteri(GL_TEXTURE_2D_ARRAY, GL_TEXTURE_MAX_LEVEL, config.levels - 1);
 
+  if (g_ogl_config.bSupportsTextureStorage)
+  {
+    glTexStorage3D(GL_TEXTURE_2D_ARRAY, config.levels, GL_RGBA8, config.width, config.height,
+                   config.layers);
+  }
+
   if (config.rendertarget)
```

Comment:
What do you think on always calling glTexImage3D here if texture_storage isn't supported? So we can use glTexSubImage3D unconditionally afterwards.